src/promptflow/promptflow/_sdk/operations/_local_azure_connection_operations.py

```diff
@@ -16,20 +17,38 @@
 
 class LocalAzureConnectionOperations(TelemetryMixin):
     def __init__(self, connection_provider, **kwargs):
-        from azure.identity import DefaultAzureCredential
 
         from promptflow.azure._pf_client import PFClient as PFAzureClient
 
         super().__init__(**kwargs)
         subscription_id, resource_group, workspace_name = self._extract_workspace(connection_provider)
         self._pfazure_client = PFAzureClient(
             # TODO: disable interactive credential when starting as a service
-            credential=DefaultAzureCredential(exclude_interactive_browser_credential=False),
+            credential=self.get_credential(),
             subscription_id=subscription_id,
             resource_group_name=resource_group,
             workspace_name=workspace_name,
         )
 
+    @classmethod
+    def get_credential(cls):
+        from azure.identity import DefaultAzureCredential, DeviceCodeCredential
+
+        if is_from_cli():
```

Comment:
confirm: extension calling us is on cli path?

Reply:
confirmed. flow test/run create calls via cli command, as extension now get connection from PFS, they are not able to get the secrets required to execute flow.